src/bin/signal_handler.rs

```diff
@@ -62,23 +69,48 @@ mod imp {
                         info!("{}", v)
                     }
                 }
-                SIGUSR2 => {
-                    if backup_path.is_empty() {
-                        info!("empty backup path, backup is disabled");
-                        continue;
-                    }
-
-                    info!("backup db to {}", backup_path);
-                    if let Err(e) = engine.backup_at(backup_path) {
-                        error!("fail to backup: {}", e);
-                    }
-                    info!("backup done");
-                }
+                SIGUSR2 => profile_memory(&profiling_memory),
                 // TODO: handle more signal
                 _ => unreachable!(),
             }
         }
     }
+
+    fn profile_memory(flag: &Arc<AtomicBool>) {
+        if flag.load(Ordering::SeqCst) {
+            warn!("last memory profiling has not finished yet.");
+            return;
+        }
+        unsafe {
+            if let Err(e) = jemallocator::mallctl_set(b"prof.dump\0", DUMP_FILE_NAME) {
```

Comment:
Because no std-related things should be included in an allocator crate. So there is no good way to verify if a str is a valid str. So using `&[u8]` is more suitable here, so that caller can construct a valid cstring simply follow the format b"xxx\0" or construct a cstring explicitly. I prefer the `\0` way because it's clean and simple.